src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -149,13 +157,28 @@ class RsExtractFunctionConfig private constructor(
             val letBindings = fn.descendantsOfType<RsPatBinding>()
                 .filter { it.textOffset <= end }
 
+            val implLookup = ImplLookup.relativeTo(fn)
             val parameters = letBindings
-                .filter { it.textOffset <= start }
-                .filter {
-                    ReferencesSearch.search(it, LocalSearchScope(fn))
-                        .any { ref -> ref.element.textOffset in start..end }
+                .mapNotNull {
+                    if (it.textOffset > start) return@mapNotNull null
+                    val result = ReferencesSearch.search(it, LocalSearchScope(fn))
+                    if (result.none { it.element.textOffset in start..end }) return@mapNotNull null
+
+                    val requiredBorrowing = result.any { it.element.textOffset > end }
```

Comment:
`result.any { it.element.textOffset > end }` condition is not enough here
In the following case we still can pass variable by reference
```rust
fn foo() {
    let a = Vec::new();
    <selection>f(&a);</selection>
}

fn f(a: &Vec<i32>) {}
```
but pass it by value
```rust
fn foo() {
    let a = Vec::new();
    bar(a);
}

fn bar(a: Vec<i32>) {
    f(&a);
}

fn f(a: &Vec<i32>) {}
```